Clarifying IP hint behavior

mstojens opened the pull request:
Added some text to address #2 and #5 
- reference to SVCB draft with regards to hints
- explained why resolvers should provide hints
- clients MUST subject resolved IP addresses to the same security requirements as other IP address

bemasc commented:
I think we should invert this advice.  We should recommend _not_ including IP hints, and instead instruct recursive resolvers to append the relevant A and AAAA records in the Additional Section.

The IP hints are strictly a performance hack when using a legacy _non_-SVCB-aware recursive resolver.  This text is specifically applicable only to SVCB-aware recursive resolvers, so we can tell them to do it the right way instead.

tfpauly commented:
@bemasc 's suggestion sounds good to me. Let's emphasize that the addresses should come in the same response as the SVCB record, with a preference for Additional Section.

mglt commented:
I reviewed the draft on the git and came with a similar comment.

preventing the use of ipv4hint

My first comment is that we should favor IPv6 over IPv4 in our example. 

My second comment is that the first example uses a local target that is using a hint mentioned as SHOULD NOT in the svcb specification. The scenario I see that might be useful are: 
 * From the IP address of the unencrypted resolver: when the encrypted resolver has a different IP address than the one advertised.
  * From the name of encrypted resolvers. 
  
Duplication of the IP address adds operational complexity and is a source of misconfiguration and makes updates harder. This may result in the resolver being unreachable and I do not think we should take that path even with a low probability.
If I try to balance this with the potential outcomes, it at most reduces the connectivity of an additional round trip. Firstly, this does not sound - at least to me - as a big issue. Then it seems to me the same results can be achieved safely using additional data RRSets. If the additional round trip is performed, it is also likely that resolutions will prefetch the first connection performed by the end user. In any case, I do not see the benefit provided by the ip hint to compete with its disadvantages.

mstojens commented:
I have updated the text accordingly. The description of the need for an additional name resolution still mentions this will result if there are no A/AAAA or IP hints included because it's true, but the SHOULD is for the A/AAAA. I have also removed the IP hints from the examples altogether.

mstojens commented:
This PR also address the duplicate issue #10 regarding adding ipv6 hints.

mglt commented:
sure, thanks for the update.

On Thu, Jun 3, 2021 at 9:23 PM Tommy Jensen ***@***.***>
wrote:

> This PR also address the duplicate issue #10
> <https://github.com/ietf-wg-add/draft-ietf-add-ddr/issues/10> regarding
> adding ipv6 hints.
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/ietf-wg-add/draft-ietf-add-ddr/pull/12#issuecomment-854288284>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABSL7SVW45XAE6HZBKCXFRDTRATJXANCNFSM4ZUBYGHA>
> .
>


-- 
Daniel Migault
Ericsson
8400 boulevard Decarie
Montreal, QC   H4P 2N2
Canada

Phone: +1 514-452-2160
